src/libs/Navigation/AppNavigator/AuthScreens.js

```diff
@@ -142,17 +143,19 @@ class AuthScreens extends React.Component {
         NameValuePair.get(CONST.NVP.PRIORITY_MODE, ONYXKEYS.NVP_PRIORITY_MODE, 'default');
         NameValuePair.get(CONST.NVP.IS_FIRST_TIME_NEW_EXPENSIFY_USER, ONYXKEYS.NVP_IS_FIRST_TIME_NEW_EXPENSIFY_USER, true);
 
-        API.Get({
-            returnValueList: 'nameValuePairs',
-            nvpNames: ONYXKEYS.NVP_PREFERRED_LOCALE,
-        }).then((response) => {
-            const preferredLocale = lodashGet(response, ['nameValuePairs', 'preferredLocale'], CONST.DEFAULT_LOCALE);
-            if ((currentPreferredLocale !== CONST.DEFAULT_LOCALE) && (preferredLocale !== currentPreferredLocale)) {
-                setLocale(currentPreferredLocale);
-            } else {
-                Onyx.set(ONYXKEYS.NVP_PREFERRED_LOCALE, preferredLocale);
-            }
-        });
+        if (Permissions.canUseInternationalization(this.props.betas)) {
```

Comment:
I think this is a race condition for a fresh login. How can we guarantee that `this.props.betas` is set? We are calling `User.getBetas()` _after_ so surely this won't work.

Reply:
Same comment here, we don't need to check for the betas here, let's only check when presenting the UI to change the language.

Reply:
That's an edge case we don't really care about in any scenario other than us testing things, so I think it's fine.

Reply:
wait, are you saying the issue we are fixing is an edge case? 😕